backend/dal/Services/Concrete/ParcelService.cs

```diff
@@ -200,25 +201,106 @@ public Parcel Add(Parcel parcel)
         /// <returns></returns>
         public Parcel Update(Parcel parcel)
         {
-            parcel.ThrowIfNotAllowedToEdit(nameof(parcel), this.User, Permissions.PropertyEdit);
+            var existingParcel = this.Context.Parcels
+                .Include(p => p.Agency)
+                .Include(p => p.Address)
+                .Include(p => p.Evaluations)
+                .Include(p => p.Buildings).ThenInclude(b => b.Evaluations)
+                .Include(p => p.Buildings).ThenInclude(b => b.Address)
+                .SingleOrDefault(u => u.Id == parcel.Id) ?? throw new KeyNotFoundException();
```

Comment:
Probably should change to `FirstOrDefault(...)`

Reply:
Hmm, I don't know, I kind of like the documented behaviour of this when we are searching for a unique key:

Returns the only element of a sequence, or a default value if the sequence is empty; this method throws an exception if there is more than one element in the sequence.

seems appropriate since there should never be more than one match.

Reply:
At the end of the day it doesn't change much here.  As the LINQ statement becomes an SQL statement and if it returns more than one it would throw an error.  However since you're making a request for the primary key there never will be more than one or none returned.  So using `Single` technically is doing more work than is required.